yt_dlp/extractor/rtrfm.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class RTRFMIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rtrfm\.com\.au/(?:shows|show-episode)/(?P<id>[^/?\#&]+)'
+    _TESTS = [
+        {
+            'url': 'https://rtrfm.com.au/shows/breakfast/',
+            'md5': '46168394d3a5ce237cf47e85d0745413',
+            'info_dict': {
+                'id': 'breakfast-2021-11-16',
+                'ext': 'mp3',
+                'series': 'Breakfast with Taylah',
+                'title': r're:^Breakfast with Taylah \d{4}-\d{2}-\d{2}$',
+                'description': 'md5:0979c3ab1febfbec3f1ccb743633c611',
+            },
+            'skip': 'ID and md5 changes daily',
+        },
+        {
+            'url': 'https://rtrfm.com.au/show-episode/breakfast-2021-11-11/',
+            'md5': '396bedf1e40f96c62b30d4999202a790',
+            'info_dict': {
+                'id': 'breakfast-2021-11-11',
+                'ext': 'mp3',
+                'series': 'Breakfast with Taylah',
+                'title': 'Breakfast with Taylah 2021-11-11',
+                'description': 'md5:0979c3ab1febfbec3f1ccb743633c611',
+            },
+        },
+        {
+            'url': 'https://rtrfm.com.au/show-episode/breakfast-2020-06-01/',
+            'md5': '594027f513ec36a24b15d65007a24dff',
+            'info_dict': {
+                'id': 'breakfast-2020-06-01',
+                'ext': 'mp3',
```

Comment:
```suggestion
                'ext': 'mp4',
```
test was failing

Reply:
After some experimentation I have determined that .mp4 in the URL returned from the restreams server always means an error, either the file for that date has expired, or the show did not occur on that date (for eg a weekday-only show).

In addition there is no other indicator of the error, such as in the JSON HTTP response status code (always 200) or in the downloaded JSON containing the mp4 URL.

For a current show the .mp4 version of the URL returns HTTP 403 and the .mp3 URL returns HTTP 200. For an expired episode or wrong date for the show, the .mp4 URL returns HTTP 404 and the .mp3 URL returns HTTP 403.

I'm not sure what to do with this info. I see there is a way to tell the user what the error was (added), but is there a way for the test cases to expect an error?

Reply:
skipping the test is fine